src/background/modules/OmniboxSearch.js

```diff
@@ -63,33 +37,85 @@ function openTabUrl(url, disposition) {
+    if (!frequently) {
+        return [];
     }
 
-    const suggestions = searchResult.map((emoji) => {
+    return await Promise.all(frequently.map(async (emoji) => {
+        const chosenSkin = await getCurrrentEmojiSkinFromEmoji(emoji);
         return {
-            description: browser.i18n.getMessage("searchResultDescription", [
-                emoji.native,
+            description: browser.i18n.getMessage("searchResultDescriptionFrequently", [
+                chosenSkin.native,
                 emoji.name,
-                emoji.colons
+                chosenSkin.shortcodes,
+                browser.i18n.getMessage("extensionName")
             ]),
-            content: emoji.native
+            content: chosenSkin.native
         };
-    });
+    }));
+}
 
+/**
+ * Trigger the evaluation for the search for emojis.
+ *
+ * @public
+ * @param {string} text The text entered by the user.
+ * @param {function(import("webextension-polyfill").Omnibox.SuggestResult[]): void} suggest Callback to provide suggestions.
+ * @see {@link https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/omnibox/onInputChanged}
+ */
+export async function triggerOmnixboxSuggestion(text, suggest) {
+    text = text.trim();
+    const reloadCachedSettingsPromise = EmojiMartDataStore.reloadCachedSettings();
+
+    /** @type {import("/common/modules/EmojiSearched.d.ts").EmojiSearched[]?} */
+    const searchResult = await (await getEmojiMart()).SearchIndex.search(text);
+    console.debug(`triggerOmnixboxSuggestion (searching for "${text}"), result:`, searchResult);
+
+    // Ensures the skin settings are up-to-date by waiting for data refresh
+    await reloadCachedSettingsPromise;
+
+    /** {@type import("webextension-polyfill").Omnibox.SuggestResult[]} */
+    let suggestions = [];
+    if (searchResult) {
+        suggestions = await Promise.all(searchResult?.map(async (emoji) => {
+            // This falls back to the default skin if the current skin is not available
+            const chosenSkin = await getCurrrentEmojiSkinFromEmoji(emoji);
+            return {
+                description: browser.i18n.getMessage("searchResultDescription", [
+                    chosenSkin.native,
+                    emoji.name,
+                    // NOTE: This uses the base skin, because the skin tone modifier as a shortcode is not useful UX-wise to display
+                    emoji.skins[0].shortcodes
+                ]),
+                content: chosenSkin.native
+            };
+        }));
+    }
+
+    const emojiSearch = await AddonSettings.get("emojiSearch");
+    /** {@type number} */
+    const maximumSuggestions = emojiSearch.maximumResults || 0;
+    if (emojiSearch.enableFillingResults || text == "") {
+        if (suggestions.length < maximumSuggestions || maximumSuggestions === 0) {
+            const frequentlyUsedSuggestions = await getFrequentlyUsedAsSuggestions(maximumSuggestions || 10);
```

Comment:
Problem/bug: This does not use up2date frequent emojis, somehow still somewhat/somewhere cached, apparently?

Reply:
https://github.com/rugk/awesome-emoji-picker/issues/153